src/RedisResourceManager.php

```diff
@@ -315,8 +371,10 @@ protected function connect(array &$resource)
             }
 
             $resource['initialized'] = true;
-            if ($resource['password']) {
-                $redis->auth($resource['password']);
+            if ($resource['user'] && $resource['password']) {
+                $redis->auth([$resource['user'], $resource['password']]);
+            } elseif ($resource['password']) {
+                $redis->auth([$resource['password']]);
```

Comment:
we could've kept the original implementation as `string` still works for `password`-only.

Reply:
_(not a task, just FYI)_